src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -422,22 +442,50 @@ public Object onTraversalSuccess() {
      * Method to create feature readers for input files or GCS URLs
      * in the current batch
      *
-     * @param sampleNames  List of all sample names
      * @param sampleNameToFileName  Sample name to file name mapping
      * @param batchSize  Current batch size
-     * @param sampleCount  total samples in this import
      * @param lowerSampleIndex  0-based Lower bound of sample index -- inclusive
      * @return  Feature readers to be imported in the current batch
      */
-    private Map<String, FeatureReader<VariantContext>> getFeatureReaders(final List<String> sampleNames, final Map<String, String> sampleNameToFileName,
-                                                                         final int batchSize, final int sampleCount, final int lowerSampleIndex) {
+    private Map<String, FeatureReader<VariantContext>> getFeatureReaders(final LinkedHashMap<String, Path> sampleNameToFileName,
+                                                                         final int batchSize, final int lowerSampleIndex) {
         final Map<String, FeatureReader<VariantContext>> sampleToReaderMap = new LinkedHashMap<>();
+        logger.debug("Starting batch header preload: "+ System.currentTimeMillis());
+        final List<Future<FeatureReader<VariantContext>>> futures = new ArrayList<>();
+        final List<String> sampleNames = new ArrayList<>(sampleNameToFileName.keySet());
+        for(int i = lowerSampleIndex; i < sampleNameToFileName.size() && i < lowerSampleIndex+batchSize; ++i) {
+            final String sampleName = sampleNames.get(i);
+            futures.add(headerLoadingExecutorService.submit(() -> {
+                final Path variantPath = sampleNameToFileName.get(sampleName);
+                try {
+                    return new InitializedQueryWrapper(getReaderFromPath(variantPath), intervals.get(0));
+                } catch (final IOException e) {
+                    throw new UserException.CouldNotReadInputFile("Couldn't read file: " + variantPath.toUri(), e);
+                }
+            }));
+        }
+
+        futures.forEach(f -> {
```

Comment:
That sounds neat but I'm not going to do it right now.  I don't think it would be a significant improvement since the processing is minimal per future so I don't think we're wasting much time.